tests/PySys/monitoring/monitoring_with_collectd/run.py

```diff
@@ -0,0 +1,124 @@
+from pysys.basetest import BaseTest
+
+import time
+import re
+import json
+
+"""
+Validate tedge-dm-agent  messages that are published
+on tedge/measurements
+
+Given a configured system
+When we start the collectd with sudo in the background
+When we start the tedge-dm-agent with sudo in the background
+When we start tedge sub with sudo in the background
+Wait for couple of seconds to publish couple of batch of messages
+Then we kill tedge sub with sudo as it is running with a different user account
+Then we validate the  messages in the output of tedge sub,
+"""
+
+
+class MonitoringWithCollectd(BaseTest):
+
+    def setup(self):
+        self.js_msg = ""
+        self.cpu_cnt = 0
+        self.memory_cnt = 0
+        self.time_cnt = 0
+        self.disk_cnt = 0
+
+    def execute(self):
+        tedge = "/usr/bin/tedge"
+        sudo = "/usr/bin/sudo"
+
+        collectd = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "collectd"],
+            stdouterr="collectd",
+        )
+
+        collectd_mapper = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "tedge-dm-agent"],
+            stdouterr="collectd_mapper",
+        )
+
```

Comment:
As mentioned in a previous comment, starting collectd and dm-agent can be done in the `setup` method and also register a cleanup function in that `setup` method itself to stop these once the test is complete.

Reply:
Sure